File: LilacForum-back/LilacForum/src/main/java/com/lilac/pojo/Notification.java
package com.lilac.pojo;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Notification {

    private Integer id; // 主键ID
    private Integer userId; // 接收通知的用户ID
    private String type; // 通知类型（如好友请求、消息通知等）
    private String content; // 通知内容
    private Boolean isRead = false; // 是否已读
    private Timestamp createdAt; // 通知的创建时间

}

File: LilacForum-front/LilacForum/src/interface/Notification.ts
// src/interface/Notification.ts
export interface Notification {
    id: number;             // 通知ID
    userId: number;         // 接收通知的用户ID
    type: NotificationType; // 通知类型
    content: string;        // 通知内容
    isRead: boolean;        // 是否已读
    createdAt: string;      // 通知的创建时间
  }
  
  // 通知类型定义
  export type NotificationType = 'FRIEND_REQUEST' | 'MESSAGE' | 'OTHER';
  